### ISSUE TITLE
Forbid to raise exceptions that are caught in except cases

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#742]
# Rule request

## Thesis

We need to disallow code like this:

```python
try:
    print(1)
    raise ValueError()
except ValueError:
    print(2)
```

This is just a `goto` statement.

Related:

```
Finding except clauses that raise a different exception class than they catch:

$ astpath "//ExceptHandler[body//Raise/exc//Name and not(contains(body//Raise/exc//Name/@id, type/Name/@id))]" | head -5
./hashlib.py:144        >except ImportError:
./plistlib.py:89        >        except KeyError:
./plistlib.py:103       >        except KeyError:
./nntplib.py:868        >        except ValueError:
./argparse.py:1116      >        except KeyError:
```

From: https://github.com/hchasestevens/astpath

## Reasoning

We do not need a `goto` in our code.

**Comment by sobolevn** [comment]
Nope, this is not going to be implemented for two reasons:
1. We still can't catch things like `raise IndexError()` and `except LookupError:` in one of the cases
2. Sometimes this `goto` is useful

Sorry.
